expyfun/stimuli/_tracker.py

```diff
@@ -274,13 +307,35 @@ def respond(self, correct):
                     reversals=[int(s) for s in self._reversals],
                     x=[float(s) for s in self._x])))
 
+    def check_valid(self, n_reversals):
+        """If last reversals contain reversals exceeding x_min or x_max.
+
+        Parameters
+        ----------
+        n_reversals : int
+            Number of reversals (starting from the end to check).
+
+        Returns
+        -------
+        valid : bool
+            True if none of the reversals are at x_min or x_max and False
+            otherwise.
+        """
+        idx = np.where([r != 0 for r in self._reversals])[0]
+        self._valid = not any([x in self._bad_reversals
+                               for x in self._reversals[idx[-n_reversals:]]])
```

Comment:
What if we make `self._bad_reversals` a boolean array of the same length as `self._reversals` (setting True when there is a bad one), then you can just do:
```
self._valid = not self._bad_reversals[self._reversals != 0][-n_reversals:].any()
```
Is it clearer for you, too?